File: src/main/clojure/graphbrain/web/views/page.clj
;   Copyright (c) 2016 CNRS - Centre national de la recherche scientifique.
;   All rights reserved.
;
;   Written by Telmo Menezes <telmo@telmomenezes.com>
;
;   This file is part of GraphBrain.
;
;   GraphBrain is free software: you can redistribute it and/or modify
;   it under the terms of the GNU Affero General Public License as published by
;   the Free Software Foundation, either version 3 of the License, or
;   (at your option) any later version.
;
;   GraphBrain is distributed in the hope that it will be useful,
;   but WITHOUT ANY WARRANTY; without even the implied warranty of
;   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
;   GNU Affero General Public License for more details.
;
;   You should have received a copy of the GNU Affero General Public License
;   along with GraphBrain.  If not, see <http://www.gnu.org/licenses/>.

(ns graphbrain.web.views.page
  (:use hiccup.core
        hiccup.page))

(defn- version
  [dev]
  (if dev
    (str "?" (rand-int 999999999))
    ""))

(defn css+js
  [dev]
  (str
   "<link href='/css/bootstrap.min.css' type='text/css' rel='Stylesheet' />"
   "<link href='/css/gb.css" (version dev) "' type='text/css' rel='Stylesheet' />"
   "<script src='/js/jquery-1.11.1.min.js' type='text/javascript'></script>"
   "<script src='/js/bootstrap.min.js' type='text/javascript' ></script>"
   "<script src='/js/gbui.js" (version dev) "' type='text/javascript' ></script>"))

(defn page [& {:keys [title body-fun js dev]}]
  (html5 {:lang "en"}
         [:head
          [:meta {:charset "utf-8"}]
          [:meta {:viewport "width=device-width, initial-scale=1.0"}]
          [:meta {:description ""}]
          [:meta {:author ""}]
          [:title (str title " - Graphbrain")]
          (css+js dev)
          [:script {:src "http://html5shim.googlecode.com/svn/trunk/html5.js"}]]
         [:body
          (body-fun)
          [:script {:language "javascript"} js]]))

File: gb/ui/page.py
#   Copyright (c) 2016 CNRS - Centre national de la recherche scientifique.
#   All rights reserved.
#
#   Written by Telmo Menezes <telmo@telmomenezes.com>
#
#   This file is part of GraphBrain.
#
#   GraphBrain is free software: you can redistribute it and/or modify
#   it under the terms of the GNU Affero General Public License as published by
#   the Free Software Foundation, either version 3 of the License, or
#   (at your option) any later version.
#
#   GraphBrain is distributed in the hope that it will be useful,
#   but WITHOUT ANY WARRANTY; without even the implied warranty of
#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#   GNU Affero General Public License for more details.
#
#   You should have received a copy of the GNU Affero General Public License
#   along with GraphBrain.  If not, see <http://www.gnu.org/licenses/>.


def html(title, body):
    return """
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="icon" href="/static/img/favicon.png">

    <title>%s - GraphBrain</title>

    <link href="/static/css/bootstrap.min.css" rel="stylesheet">
    <link href="/static/css/graphbrain.css" rel="stylesheet">
  </head>

  <body>
    <nav class="navbar navbar-default navbar-static-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar"
          aria-expanded="false" aria-controls="navbar">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-left" href="/"><img src="/static/img/GB_logo.png" height="50px"></a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
            <li><a href="/">Home</a></li>
            <li><a href="/parser">Parser</a></li>
          </ul>
          <form class="navbar-form navbar-right" action="/search" method="get">
            <div class="form-group">
              <input type="text" placeholder="Search" class="form-control" name="query">
            </div>
            <button type="submit" class="btn btn-success">Search</button>
          </form>
        </div><!--/.navbar-collapse -->
      </div>
    </nav>

    %s

    <script src="/static/js/jquery-1.12.4.min.js"></script>
    <script src="/static/js/bootstrap.min.js"></script>
  </body>
</html>
    """ % (title, body)
